packages/core/src/server/socketServer.ts

```diff
@@ -166,6 +172,29 @@ export class SocketServer {
       return null;
     }
 
+    // web-infra-dev/rspack#6633
+    // when initial-chunks change, reload the page
+    // e.g: ['index.js'] -> ['index.js', 'lib-polyfill.js']
+    const newEntryPointsChunks: string[] = [];
+    for (const entrypoint of Object.values(stats.entrypoints ?? {})) {
```

Comment:
Add `if` can have better perf in this case

```suggestion
    if (stats.entrypoints) {
      for (const entrypoint of Object.values(stats.entrypoints)) {
```